### PR TITLE
cleanup workflows

### PR DESCRIPTION
**Pull request by Adammatthiesen**
This PR, merges some workflows and cleans up others.

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
So far LGTM, i'm just wondering if you test the `changeset` action cause you added a lot of extra steps to it.

**Comment by Adammatthiesen** [comment]
> So far LGTM, i'm just wondering if you test the `changeset` action cause you added a lot of extra steps to it.

I mean technically i just added a second job to the main action.  It should still work since the second job has a very specific rule-set for it.  But will be interesting to know if i am crazy or not...

File: .github/workflows/changeset.yml [hunk 1]
```diff
@@ -1,10 +1,25 @@
-name: Release
+name: Changeset Release and Snapshot
 
 on:
   workflow_dispatch:
+    inputs:
+      runRealse:
```
Comment (Fryuni) [inline .github/workflows/changeset.yml:6]:
```suggestion
      runRelease:
```

File: .github/workflows/changeset.yml [hunk 2]
```diff
@@ -26,6 +41,7 @@ permissions:
 jobs:
   release:
     name: Release
+    if: ${{ github.event.inputs.runRealse == 'true' || github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```
Comment (Fryuni) [inline .github/workflows/changeset.yml:44]:
```suggestion
    if: ${{ github.event.inputs.runRelease == 'true' || github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

File: .github/workflows/changeset.yml [hunk 3]
```diff
@@ -55,8 +71,148 @@ jobs:
...
+
+      - run: git fetch origin main:main
+
+      - name: Setup Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Install pnpm
+        run: npm i pnpm@9.5.0 -g
```
Comment (Fryuni) [inline .github/workflows/changeset.yml:136]: I think it would be better enable corepack so it uses the exact version from package.json

Comment (Adammatthiesen) [reply .github/workflows/changeset.yml:136]: There are a few places that would need to happen, but i am not against this, i just haven't had to much experience setting up corepack within GH actions.  I will note the version in the `package.json` is currently the same.

Comment (Adammatthiesen) [reply .github/workflows/changeset.yml]: [4093ecb](https://github.com/astrolicious/studiocms/pull/218/commits/4093ecb8988a3f35df3e26af1878dbc5187610e9) done using the pnpm action that references the packageManager option! (i didnt know this action step existed... i like this...)

**Review by Fryuni — APPROVED** [review]

**Comment by Adammatthiesen** [comment]
!coauthor

**Pull request merged**